Pluto-tv/acyl
# Feature On Branches Acyl Config Option

Currently Acyl Config only provides the means to create ephemeral environments based on PRs into Target Branches. This change allows for tighter fine-grain controls over the PR Branch name specifications using regex to determine when the branch name matches one of many defined definitions.

```yaml
on_branches:
  - feature/*
  - dev/JIRA-1234/*
  - release/foo/*
  - awesome-tests
```
Changes
---
GITHUB EVENT
- update check relevancy PR with On Branch check on target branch match when provided
- add pass/fail tests for github event On Branch
- update QAType model
- update nitro RepoConfig model
- update logic for relevancy PR with On Branch
- enhance tests
- add test for varied on branch options

ACYL CONFIG
- update acyl config to utilize on branch feature
